src/utilities/idf/IdfObject_Impl.hpp

```diff
@@ -101,6 +101,11 @@ namespace detail {
      */
     virtual boost::optional<std::string> getString(unsigned index, bool returnDefault = false, bool returnUninitializedEmpty = false) const;
 
+    /** Like getString except for reference fields getString will return the name of the referenced object.
+     * This method, getField, will always return the string value of the field.
+     * For IdfObject, this is the same as getString */
+    virtual boost::optional<std::string> getField(unsigned index, bool returnDefault = false) const;
```

Comment:
`getField` was only a WorkspaceObject method before. Now it's on IdfObject for the public side, IdfObject_Impl on the impl side and overriden in WorkspaceObject_Impl.

This allows IdfExtensibleGroup, since it uses a std::shared_ptr<IdfObject_Impl>, to use polymorphism to "do the right thing(TM)"